core/src/main/java/org/apache/druid/math/expr/ExprEval.java

```diff
@@ -48,7 +49,7 @@
   public static ExprEval deserialize(ByteBuffer buffer, int position)
   {
     // | expression type (byte) | expression bytes |
-    ExprType type = ExprType.fromByte(buffer.get(position));
+    ExprType type = ExprType.fromByte((byte) (buffer.get(position) & TYPE_MASK));
```

Comment:
I think it would be better to add comment suggesting that only `BufferLambdaExpressionAggregator` calls this hence we are clearing the sign bit due to implementation in aggregator.

Reply:
I reworked this to do the masking in the buffer aggregator instead of here